lib/tabry/models/flags_list.rb

```diff
@@ -19,11 +19,20 @@ def first_required_flag(used:)
       def options(token, used:)
         to_a.map do |flag|
           if token&.start_with?("-") && flag.name_with_dashes.start_with?(token) && !used[flag.name]
-            flag.name_with_dashes
+            output_string(flag)
           end
         end.compact
       end
 
+      def output_string(flag)
+        fish_mode = ENV.fetch("TABRY_FISH_MODE", false)
```

Comment:
I'm not crazy about the idea of a ENV var here ... not that I haven't had to resort to things in other places, but still, I don't like the duplication of checking that and doing escaping. Also, you moved escaping here from the shell wrappers, but there are other models with with `options` methods, e.g. ConstOption.

What if we made `options` take another parameter (for ease of implementing, it can be optional) -- call it `params` (all the other good names are taken) -- which could affect what the `options` returns. In this case, the Bash/Fish wrapper would pass in {fish: true} to OptionsFinder (this currently goes thru Tabry::Runner.options, which is just a small wrapper function -- that could be modified to include the extra parameter too). OptionsFinder would then look something like this:

```
class OptionsFinder
  attr_reader :result, :params

  def self.options(result, token, params)
    new(result, params).options(token)
  end

  def initialize(result, params)
    @result = result
    @params = params
    # note -- currently we pass token along everywhere, we could probably add it here to avoid that
  end
```

In the places you want descriptions (although eventually, if we use params for more things, we'd probably want it everywhere) you can add params to the call to options(token).

The Fish Wrapper would then be responsible for turning the arrays into the proper strings. Well, I guess you're still using the Bash Wrapper, so you can still have the environment variable and check it there, e.g. it would pass in params as
```
{fish: %w[false 0].include?(ENV.fetch("TABRY_FISH_MODE", "false" )}
```
It would also have to turn the arrays (assuming options() in fish mode returns [option, description]) into the tab-separated strings, but that should be easy:
```
options.map do |(opt, desc)|
  [opt.shellescape, desc && "# #{desc}"].compact.join("\t")
end
```

Reply:
Ok, so I took out the env var, and added a fish-specific wrapper/tabry-fish executable. I like the separation now, since there is less threading of the env variable.

The only downside is that I had to add a ```<cmd> completion-fish``` command which parallels ```<cmd> completion```, but instead calls the fish wrapper. It's kind of ugly cause now it shows up in the tab completion for hello:
```
> examples/all_in_one/hello completion "hello " 6
completion
completion-fish
things
completion
completion-fish
things
```

is that okay?